helm/grafana/templates/postgresql/cluster.yaml

```diff
@@ -0,0 +1,76 @@
+{{- if .Values.postgresql.enabled -}}
+apiVersion: postgresql.cnpg.io/v1
+kind: Cluster
+metadata:
+  labels:
+    observability.giantswarm.io/tenant: giantswarm
+  name: {{ .Values.postgresql.clusterName }}
+  namespace: {{ .Release.Namespace }}
+spec:
+  {{- if .Values.postgresql.backup.name }}
+  backup:
+    barmanObjectStore:
+      data:
+        compression: gzip
+      {{- if eq .Values.global.provider.kind "aws" }}
+      destinationPath: {{ include "chart.s3.backupPath" . }}
+      s3Credentials:
+        inheritFromIAMRole: true
+      {{- else if eq .Values.global.provider.kind "azure" }}
+      destinationPath: {{ include "chart.azure.backupPath" . }}
+      azureCredentials:
+        storageAccount:
+          name: {{ include "chart.storageBucket.name" . }}
+          key: accountName
+        storageKey:
+          name: {{ include "chart.storageBucket.name" . }}
+          key: accountKey
+      {{- end }}
```

Comment:
We will probably have a third case, on on-prem installations for instance, where customers will provide their own object storage, and we can't make any assessment on the kind of object storage.
For these cases, I think we need to support a "freeform" object store definition.